### ISSUE TITLE
unsupported flash type: 85601385

### ISSUE THREAD

**Issue by Aginarikar** [BertoldVdb/jms578flash#7]
Hello I got a similar issue to issue #6

My unsupported flash type (number/ID) looks similar, but it is not the same: **85601385**
Can it be found/added please?

**Comment by BertoldVdb** [comment]
Hello,

Do you know the number written on the chip? It will make it easier.

Sincerely,
Bertold

Op ma 4 sep. 2023 14:59 schreef Aginarikar ***@***.***>:

**Comment by Aginarikar** [comment]
Yes, after looking closer the chip is:
**PUYA P25D40H**
the second line reads:
9C1EC1D
just like in this found image:
https://52audio-images.oss-cn-shenzhen.aliyuncs.com/wp-content/uploads/2021/04/2021042112180717.jpg

**Comment by BertoldVdb** [comment]
Hello,

Super. I will try to add it tonight or tomorrow.

Sincerely,
Bertold

Op ma 4 sep. 2023 15:36 schreef Aginarikar ***@***.***>:

**Comment by BertoldVdb** [comment]
Chip is added.

**Comment by Aginarikar** [comment]
Thank you very much for this effort.
I managed to extract the firmware (without involving bootrom), then dump the bootrom.
When trying to flash the firmware with bootrom option, I got the same error message as in issue #6 .

Trying to flash the firmware without bootrom option was done sucessfully.
(Regarding JMS578 vs JMS579, lsusb previous indicated product 0578 with name JMS579 (!), now it reports product 0579 with still the same name JMS579. (And update-usbids I understand is now obsolete.))
So the issue can be closed. 

Going off topic and hoping for someone to put me in the right direction if any, I'm still experiencing about the same errors as before with the unit. Connecting different SSDs with and without the UAS mode disable quirk as described in the below link (now updated to «0579»), file transfers stops unexpectedly and drives are unmounted unexpectedly as well. Using Ubuntu 22.04 6.2.0-32. https://leo.leung.xyz/wiki/How_to_disable_USB_Attached_Storage_(UAS)
Currently it would take a rather long time to check using an even slower computer if the SSDs themselves actually have been broken...
